src/figuresClasses.ts

```diff
@@ -1,19 +1,113 @@
+type Shapes = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shapes,
+  color: Color,
+  getArea(): number
+}
 
+function validator(key: string, value: number):void {
+  if (value <= 0) {
+    throw new Error(`Please enter valid ${key} value`);
+  }
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shapes = 'triangle';
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    color: Color,
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    validator('a', a);
+    validator('b', b);
+    validator('c', c);
+
+    const arraySides: number[] = [a, b, c]
+      .sort((number1, number2) => number1 - number2);
+
+    if (arraySides[2] >= (arraySides[1] + arraySides[0])) {
+      throw new Error('Please enter correct side valies!');
+    }
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const semiPerimetr = (this.a + this.b + this.c) / 2;
+
+    const square = Math.sqrt(semiPerimetr * (semiPerimetr - this.a)
+      * (semiPerimetr - this.b) * (semiPerimetr - this.c));
 
+    return Math.round(square * 100) / 100;
```

Comment:
you can creat a function for rounding your result, `Math.floor` for all works perfectly